pkg/reconciler/workload/heartbeat/heartbeat_manager.go

```diff
@@ -35,25 +37,26 @@ type clusterManager struct {
 
 func (c *clusterManager) Reconcile(ctx context.Context, cluster *workloadv1alpha1.WorkloadCluster) error {
 	latestHeartbeat := time.Time{}
-	for _, c := range cluster.Status.Conditions {
-		if c.LastHeartbeatTime.Time.After(latestHeartbeat) {
-			latestHeartbeat = c.LastHeartbeatTime.Time
-		}
+	if cluster.Status.LastSyncerHeartbeatTime != nil {
+		latestHeartbeat = cluster.Status.LastSyncerHeartbeatTime.Time
 	}
 	if latestHeartbeat.IsZero() {
-		// Never seen a heartbeat; for now, this means the syncer hasn't heartbeated -- in the future, this should be un-Ready.
-		return nil
-	}
-
-	if time.Since(latestHeartbeat) > c.heartbeatThreshold {
+		klog.V(5).Infof("Marking WorkloadCluster %s|%s not ready due to no heartbeat", cluster.ClusterName, cluster.Name)
+		conditions.MarkFalse(cluster,
+			workloadv1alpha1.WorkloadClusterReadyCondition,
+			workloadv1alpha1.ErrorHeartbeatMissedReason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"No heartbeat yet seen")
+	} else if time.Since(latestHeartbeat) > c.heartbeatThreshold {
+		klog.V(5).Infof("Marking WorkloadCluster %s|%s not ready due to a stale heartbeat", cluster.ClusterName, cluster.Name)
 		conditions.MarkFalse(cluster,
 			workloadv1alpha1.WorkloadClusterReadyCondition,
 			workloadv1alpha1.ErrorHeartbeatMissedReason,
 			conditionsv1alpha1.ConditionSeverityError,
 			"No heartbeat since %s", latestHeartbeat)
 	} else {
-		conditions.MarkTrue(cluster,
-			workloadv1alpha1.WorkloadClusterReadyCondition)
+		klog.V(5).Infof("Marking WorkloadCluster %s|%s ready", cluster.ClusterName, cluster.Name)
```

Comment:
wished there was `MarkTrue(...) (change bool)` to avoid non-edge log messages

Reply:
Another symptom of our controller architecture. At least it's low-level?

Reply:
Until then, why not only log this `if cluster.Status.IsReady()`?

Reply:
This log level was useful for debugging. Until we see that it's actually a problem, I'd prefer to leave it in.